src/main/java/games/chess/actions/Promotion.java

```diff
@@ -0,0 +1,79 @@
+package games.chess.actions;
+
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.swing.plaf.basic.BasicInternalFrameTitlePane.MoveAction;
+
+
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import core.interfaces.IExtendedSequence;
+import games.chess.components.ChessPiece;
+import games.chess.ChessGameState;
+import games.chess.actions.MovePiece;
+import games.chess.components.ChessPiece.ChessPieceType;
+
+public class Promotion extends AbstractAction {
+    private final int startX;
+    private final int startY;
+
+    private final int targetX;
+    private final int targetY;
+    private final ChessPieceType newPieceType;
+
+
+    public Promotion(int startX, int startY, int targetX, int targetY, ChessPieceType newPieceType) {
+        this.startX = startX;
+        this.startY = startY;
+        this.targetX = targetX;
+        this.targetY = targetY;
+        this.newPieceType = newPieceType;
+    }
+
+
+    @Override
+    public boolean execute(AbstractGameState gs) {
+        MovePiece move = new MovePiece(startX, startY, targetX, targetY);
+        if (!move.execute(gs)) {
+            return false;
+        }
```

Comment:
In this case you should definitely throw an error - or better, do this within the move.execute() method. 
If the move fails then something unexpected has happened, and it is dangerous to let this fail silently and carry on as if everything was fine.